interface/language/csv/translation_utilities.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+/* Copyright (C) 2014 Kevin Yeh <kevin.y@integralemr.com>
+ *
+ * LICENSE: This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>;.
+ *
+ * @package OpenEMR
+ * @author  Kevin Yeh <kevin.y@integralemr.com>
+ * @link    http://www.open-emr.org
+ */
+
+function find_or_create_constant($constant)
+{
+    $sqlFind = " SELECT cons_id , constant_name FROM lang_constants where BINARY constant_name = ?";
+    $result = sqlStatement($sqlFind, array($constant));
+    if ($result) {
+        $row_count = sqlNumRows($result);
+        if ($row_count == 1) {
+            $row = sqlFetchArray($result);
+            return $row['cons_id'];
+        }
+        if ($row_count > 1) {
+            error_log("Duplicate Entries for language constant:" . $constant);
+            $row = sqlFetchArray($result);
+            $retval = $row['cons_id'];
+            while ($row = sqlFetchArray($result)) {
+                $sqlDelete = " DELETE FROM lang_constants where cons_id = ? ";
+                sqlStatement($sqlDelete, array($row['cons_id']));
+                $sqlDelete = " DELETE FROM lang_definitions where cons_id = ? ";
+                sqlStatement($sqlDelete, array($row['cons_id']));
+                error_log("DELETED Definitions for duplicate constant:" . $constant . "|" . $row['cons_id']);
+            }
+            return $retval;
+        }
+        if ($row_count == 0) {
+            $sqlInsert = " INSERT INTO lang_constants (constant_name) VALUES (?)";
```

Comment:
when do this should also add entry to lang_custom to ensure the csv loader and Manage Translations features are compatible. Can do this by running it through the insert_language_log() function.

Reply:
disregard above comment since best for the csv import to not mess with lang_custom